experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/list.spec.ts

```diff
@@ -315,13 +291,22 @@ describe("List", () => {
 				const checkRhs = (left: string[], others: string[][], spreadable: boolean) => {
 					const clones = others.map((other) => setSpreadable(other.slice(), spreadable));
 					const expected = left.concat(...clones);
-					it(`${prettyCall("concat", left, others, expected)}`, () => {
-						const proxies = others.map((other) =>
-							setSpreadable(createStringList(other), spreadable),
-						);
-						const actual = left.concat(...proxies);
-						assert.deepEqual(actual, expected);
-					});
+
+					itWithRoot(
+						`${prettyCall("concat", left, others, expected)}`,
+						makeSchema((_) => _.list(_.list(_.string))),
+						[],
+						(subjects) => {
+							const proxies = others.map((other) => {
+								// To create a proxy, we first insert the each array from 'others' into
+								// the tree and then read back the resulting proxy.
+								subjects.insertAtEnd(other as any);
+								return setSpreadable(subjects[subjects.length - 1], spreadable);
+							});
+							const actual = left.concat(...proxies);
+							assert.deepEqual(actual, expected);
+						},
```

Comment:
This transformation is slightly more involved...  instead of conjuring _n_ SharedTrees to create _n_ lists to concatenate append to a regular JS array, it now creates one ST and creates the lists to concat from the same tree.